.github/workflows/smoke-test.yml

```diff
@@ -90,7 +90,12 @@ jobs:
         working-directory: python
     strategy:
       matrix:
-        python-version: [3.8, 3.12]
+        python-version:
+          # - 3.8
+          - 3.12
+        optional-numpy: ['numpy', '']
+        optional-randomgen: ['randomgen', '']
+        optional-scikit-learn: ['scikit-learn', '']
```

Comment:
We only need to test the extras we make available on the package. Instead of testing 2^3 combinations, we should only test the base install, install with the numpy extra, and install with the sklearn-extra (which is a superset of the numpy extra). 

When polars comes along, we can then also test with just the polars extra. I don't think we need to test combinations of polars with other features.

Reply:
Will do, but I'll save this for last to make sure I don't break an edge case in the course of the other work.